olp-cpp-sdk-core/tests/thread/ContinuationTest.cpp

```diff
@@ -53,145 +56,167 @@ class ContinuationTest : public testing::Test {
                           .Then([](olp::thread::ExecutionContext, int,
                                    std::function<void(int)> next) { next(3); })
-                          .Finally([&promise](Response<int> response) {
-                            promise.set_value(response.GetError());
+                          .Finally([&promise](ResponseType<int> response) {
+                            promise.set_value(response);
                           });
 
   continuation.Run();
+
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
+  auto result = future.get();
+
   continuation.CancelToken().Cancel();
 
-  EXPECT_EQ(future.get().GetErrorCode(), olp::client::ErrorCode::Cancelled);
+  EXPECT_TRUE(result);
 }
 
 TEST_F(ContinuationTest, CancelExecution) {
-  std::promise<void> promise;
+  std::promise<ResponseType<int>> promise;
   auto future = promise.get_future();
 
-  olp::client::ApiResponse<olp::client::HttpResponse, olp::client::ApiError>
-      result;
-
-  auto cont =
-      Create([](olp::thread::ExecutionContext, std::function<void(int)> next) {
-        std::this_thread::sleep_for(std::chrono::milliseconds(200));
-        next(1);
-      })
-          .Finally([&promise, &result](
-                       olp::client::ApiResponse<int, olp::client::ApiError>
-                           response) {
-            result = response.GetError();
-            promise.set_value();
-          });
-
-  cont.Run();
-  std::this_thread::sleep_for(
-      std::chrono::milliseconds(50));  // simulate some delay
+  std::promise<void> cancel_promise;
+  auto cancel_future = cancel_promise.get_future();
+
+  auto continuation = Create([&cancel_future](olp::thread::ExecutionContext,
+                                              std::function<void(int)> next) {
+                        ASSERT_EQ(cancel_future.wait_for(kMaxWaitMs),
+                                  std::future_status::ready);
+                        cancel_future.get();
+                        next(1);
+                      }).Finally([&](ResponseType<int> response) {
+    promise.set_value(std::move(response));
+  });
 
-  cont.CancelToken().Cancel();
+  continuation.Run();
 
-  EXPECT_EQ(future.wait_for(std::chrono::milliseconds(1000)),
-            std::future_status::ready);
+  continuation.CancelToken().Cancel();
+  cancel_promise.set_value();
 
-  future.get();
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
+  auto result = future.get();
 
-  EXPECT_FALSE(result.IsSuccessful());
-  EXPECT_EQ(result.GetError().GetHttpStatusCode(),
+  EXPECT_FALSE(result);
+  ASSERT_EQ(result.GetError().GetHttpStatusCode(),
             static_cast<int>(olp::http::ErrorCode::CANCELLED_ERROR));
 }
 
 TEST_F(ContinuationTest, FinallyNotSet) {
-  auto cont =
+  auto continuation =
       Create([](olp::thread::ExecutionContext, std::function<void(int)>) {});
 
-  cont.Run();
+  continuation.Run();
```

Comment:
Any checks here? This test will never fail, so it can't catch any issues

Reply:
added fail() << message